src/util/generate-pnp-map.js

```diff
@@ -232,6 +232,10 @@ async function getPackageInformationStores(
       const packageName = pkg.name;
       let packageReference = pkg.version;
 
+      if (!packageReference) {
+        packageReference = '0.0.0';
```

Comment:
@arcanis not sure where and how to fix this issue with pnp
https://circleci.com/gh/yarnpkg/yarn/28459?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
I'd try to add a `console.log(resolutions)` when [this invariant](https://github.com/yarnpkg/yarn/blob/7f419103307da7856627c4ca078460234701eb75/src/util/generate-pnp-map.js#L318) fails, and see what's the key being used. If the package isn't there at all, it means that either it wasn't part of the `seedPatterns`, or that [`getResolverEntry`](https://github.com/yarnpkg/yarn/blob/7f419103307da7856627c4ca078460234701eb75/src/util/generate-pnp-map.js#L193-L209) returned `null` for some reason.

Note that you can run the exact failing test by using the following command in `packages/pkg-tests`:

```
yarn jest -t 'it should install a single dependency without version'
```